components/family-studies/api/src/main/java/org/phenotips/studies/family/internal/StatusResponse.java

```diff
@@ -25,18 +25,129 @@
+    /**
+     * Patient can be added to a family.
+     */
+    OK(200,
+        "",
+        ""),
+
+    /**
+     * Duplicate patient in list.
+     */
+    DUPLICATE_PATIENT(400,
+        "duplicate",
+        "There is a duplicate link for patient %s"),
+
+    /**
+     * Patient cannot be added to a family because it is already associated with another family.
+     */
+    ALREADY_HAS_FAMILY(501,
+        "familyConflict",
+        "Patient %s already belongs to a different family, and therefore cannot be added to this one."),
+
+    /**
+     * Patient cannot be added to a family because current user has insufficient permissions on family.
+     */
+    INSUFFICIENT_PERMISSIONS_ON_FAMILY(401,
+        "familyPermissions",
+        "Insufficient permissions to edit the family record."),
+
+    /**
+     * Patient cannot be added to a family because current user has insufficient permissions on family.
+     */
+    INSUFFICIENT_PERMISSIONS_ON_PATIENT(401,
+        "patientPermissions",
+        "Insufficient permissions to edit the patient record."),
+
+    /**
+     * No members to add to family.
+     **/
+    FAMILY_HAS_NO_MEMBERS(402,
+        "invalidUpdate",
+        "The family has no members. Please specify at least one patient link."),
+
+    /**
+     * Invalid patient id.
+     */
+    INVALID_PATIENT_ID(404,
+        "invalidPatientId",
+        "Could not find patient %s."),
+
+    /**
+     * Invalid family id.
+     */
+    INVALID_FAMILY_ID(404,
+        "invalidFamilyId",
+        "Could not find family %s."),
+
+    /**
+     * Unknown error.
+     */
+    UNKNOWN_ERROR(500,
+        "unknown",
+        "Could not update patient records"),
+
+    /**
+     * Patient cannot be linked to proband's family.
+     */
+    PROBAND_HAS_NO_FAMILY(501,
+        "NoFamilyForProband",
+        "Patient %s cannot be linked to proband's family, because proband %s has not family.");
+
+    private int statusCode;
+
+    private String errorType;
+
+    private String messageFormat;
+
+    private String message;
+
+    StatusResponse(int statusCode, String errorType, String messageFormat)
+    {
+        this.statusCode = statusCode;
+        this.errorType = errorType;
+        this.messageFormat = messageFormat;
+    }
 
-    /** Should contain a message if there has been an error. */
-    public String message = "";
+    /**
+     * @return status code of the response
+     */
+    public int getStatusCode()
+    {
+        return this.statusCode;
+    }
 
-    /** Should containt the type of error, if an error occured. */
-    public String errorType = "";
+    /**
+     * @return error type of the response
+     */
+    public String getErrorType()
+    {
+        return this.errorType;
+    }
+
+    /**
+     * Sets the parameter for the error message.
+     *
+     * @param parameters for the error message
+     * @return reference to self
+     */
+    public StatusResponse setMessage(Object... parameters)
```

Comment:
This looks dangerous. enum objects should, in general, be immutable. I'd remove this and the getter and introduce a `public String formatMessage(Object... parameters)` method.